src/Octoshift/SecretScanningAlertService.cs

```diff
@@ -89,7 +89,11 @@ private AlertWithLocations MatchTargetSecret(AlertWithLocations source, List<Ale
             }
 
             if (source.Alert.SecretType == target.Alert.SecretType
-                && source.Alert.Secret == target.Alert.Secret)
+                && ((source.Alert.Secret == target.Alert.Secret) ||
+                    (source.Alert.Secret.Length > target.Alert.Secret.Length && source.Alert.Secret.Contains(target.Alert.Secret)) ||
+                    (source.Alert.Secret.Length < target.Alert.Secret.Length && target.Alert.Secret.Contains(source.Alert.Secret))
+                   )
+                )
```

Comment:
While I am not sure if this is correct since I am not super familiar with secret scanning stuff, but in terms of pure code review, the condition can be simplified:
```suggestion
                && (source.Alert.Secret.Contains(target.Alert.Secret) || target.Alert.Secret.Contains(source.Alert.Secret)))
```

A couple of notes:
1. `string.Contains()` is super optimized so no need to do the length check before calling it at all.
2. A == B basically means A.Contains(B) && B.Contains(A) but since we're doing A.Contains(B) || B.Contains(A) later then we can just remove the redundant equality check.